samples/client-sdk/getting-started-daily-js/index.js

```diff
@@ -336,89 +412,81 @@ class DailyCallManager {
-    micPromptOption.disabled = true;
-    micSelector.appendChild(micPromptOption);
+    // Prepare selectors by clearing existing options and adding a non-selectable prompt.
+    Object.values(selectors).forEach((selector) => {
+      selector.innerHTML = '';
+      const promptOption = new Option(
+        `Select a ${selector.id.includes('camera') ? 'camera' : 'microphone'}`,
+        '',
+        true,
+        true
+      );
+      promptOption.disabled = true;
+      selector.appendChild(promptOption);
+    });
 
-    // Populate the selectors and set currently selected devices as selected
+    // Create and append options to the selectors based on available devices.
     devices.forEach((device) => {
-      if (device.label) {
-        const option = new Option(device.label, device.deviceId);
-
-        if (device.kind === 'videoinput') {
-          cameraSelector.appendChild(option);
-          if (device.deviceId === currentCameraDeviceId) {
-            option.selected = true;
-          }
-        } else if (device.kind === 'audioinput') {
-          micSelector.appendChild(option);
-          if (device.deviceId === currentMicDeviceId) {
-            option.selected = true;
-          }
-        }
+      if (device.label && selectors[device.kind]) {
+        const isSelected =
+          selectedDevices[device.kind === 'videoinput' ? 'camera' : 'mic']
+            .deviceId === device.deviceId;
+        const option = new Option(
+          device.label,
+          device.deviceId,
+          isSelected,
+          isSelected
+        );
+        selectors[device.kind].appendChild(option);
       }
     });
 
-    // Attach event listeners to the selectors to handle device changes.
-    this.addDeviceChangeListener(cameraSelector, 'video');
-    this.addDeviceChangeListener(micSelector, 'audio');
-  }
-
-  /**
-   * Attaches a change event listener to a device selector.
-   * This listener updates the currently selected device
-   * for video or audio based on user selection.
-   *
-   * @param {HTMLElement} selector - The <select> element for choosing a device.
-   * @param {string} deviceType - The type of device, either 'video' for cameras or 'audio' for microphones.
-   */
-  addDeviceChangeListener(selector, deviceType) {
-    selector.addEventListener('change', (e) => {
-      const deviceId = e.target.value;
-      const deviceOptions = { [`${deviceType}DeviceId`]: deviceId };
-      this.call.setInputDevicesAsync(deviceOptions);
+    //Attach event listeners for device changes.
+    Object.entries(selectors).forEach(([deviceKind, selector]) => {
+      selector.addEventListener('change', async (e) => {
+        const deviceId = e.target.value;
+        const deviceOptions = {
+          [deviceKind === 'videoinput' ? 'videoDeviceId' : 'audioDeviceId']:
+            deviceId,
+        };
+        await this.call.setInputDevicesAsync(deviceOptions);
+      });
     });
   }
 
```

Comment:
this note is for the below: `updateAndDisplayParticipantCount()`

Really, we should encourage use of `participantCounts()` as that is the most accurate way of getting the true participant count and it will be accurate immediately upon join. I'm realizing though, that our API there IS a bit confusing for such a simple demo... maybe that's why you didn't use it?

Reply:
Totally. Added this task in Linear: https://linear.app/dailyco/issue/CSDK-2122/improve-participantcounts-to-give-a-total-number-of-participants-in.

Reply:
Updated as you suggested for now.